src/main/java/com/google/firebase/auth/internal/FirebaseTokenVerifier.java

```diff
@@ -184,32 +198,40 @@ public String getProjectId() {
     return projectId;
   }
 
-  public static GooglePublicKeysManager buildGooglePublicKeysManager(HttpTransport transport) {
-    return new GooglePublicKeysManager.Builder(transport, new GsonFactory())
-            .setClock(Clock.SYSTEM)
-            .setPublicCertsEncodedUrl(FirebaseTokenVerifier.CLIENT_CERT_URL)
-            .build();
-  }
-
   /** 
    * Builder for {@link FirebaseTokenVerifier}.
    */
   public static class Builder extends IdTokenVerifier.Builder {
 
-    String projectId;
-
-    GooglePublicKeysManager publicKeysManager = DEFAULT_KEY_MANAGER;
+    private String projectId;
+    private String shortName;
+    private String method;
+    private String docUrl;
+    private GooglePublicKeysManager publicKeysManager;
 
     public String getProjectId() {
       return projectId;
     }
 
-    public Builder setProjectId(String projectId) {
+    public Builder setProjectId(String issuerPrefix, String projectId) {
       this.projectId = projectId;
-
-      this.setIssuer(ISSUER_PREFIX + projectId);
       this.setAudience(Collections.singleton(projectId));
+      this.setIssuer(issuerPrefix + projectId);
+      return this;
+    }
+
+    public Builder setShortName(String shortName) {
```

Comment:
Do you mind adding JSDoc for all of these?

Reply:
Done